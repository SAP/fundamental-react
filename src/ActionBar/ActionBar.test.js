import React from 'react';
import renderer from 'react-test-renderer';
import {
  ActionBar,
  ActionBarBack,
  ActionBarHeader,
  ActionBarActions
} from './ActionBar';

describe('<ActionBar />', () => {
  const basicActionBar = (
      <ActionBar className='blue'>
          <ActionBarBack className='blue' />
          <ActionBarHeader
              className='blue'
              title={'Page Title'}
              description={'Action Bar Description'} />
          <ActionBarActions className='blue'>
              <button>Button</button>
          </ActionBarActions>
      </ActionBar>
  );

  const basicActionBarNoClass = (
      <ActionBar>
          <ActionBarBack className='blue' />
          <ActionBarHeader
              className='blue'
              title={'Page Title'}
              description={'Action Bar Description'} />
          <ActionBarActions className='blue'>
              <button>Button</button>
          </ActionBarActions>
      </ActionBar>
  );

  const mobileActionBar = (
      <ActionBar className='blue' mobile>
          <ActionBarBack />
          <ActionBarHeader
              title={'Page Title'}
              description={'Action Bar Description'} />
          <ActionBarActions>
              <button>Button</button>
          </ActionBarActions>
      </ActionBar>
  );

  const mobileActionBarWidthSet = (
      <ActionBar mobile width='500px'>
          <ActionBarBack />
          <ActionBarHeader
              title={'Page Title'}
              description={'Action Bar Description'} />
          <ActionBarActions>
              <button>Button</button>
          </ActionBarActions>
      </ActionBar>
  );

  test('create basic Action Bar', () => {
    let component = renderer.create(basicActionBar);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(basicActionBarNoClass);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('create basic mobile Action Bar', () => {
    let component = renderer.create(mobileActionBar);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(mobileActionBarWidthSet);
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
