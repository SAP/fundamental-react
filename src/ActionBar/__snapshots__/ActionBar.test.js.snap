// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionBar /> create basic Action Bar 1`] = `
<div
  className="fd-action-bar blue"
>
  <div
    className="fd-action-bar__back blue"
  >
    <button
      className="fd-button--light fd-button--compact sap-icon--nav-back"
    />
  </div>
  <div
    className="fd-action-bar__header blue"
  >
    <h1
      className="fd-action-bar__title"
    >
      Page Title
    </h1>
    <p
      className="fd-action-bar__description"
    >
      Action Bar Description
       
    </p>
  </div>
  <div
    className="fd-action-bar__actions blue"
  >
    <button>
      Button
    </button>
  </div>
</div>
`;

exports[`<ActionBar /> create basic Action Bar 2`] = `
<div
  className="fd-action-bar"
>
  <div
    className="fd-action-bar__back blue"
  >
    <button
      className="fd-button--light fd-button--compact sap-icon--nav-back"
    />
  </div>
  <div
    className="fd-action-bar__header blue"
  >
    <h1
      className="fd-action-bar__title"
    >
      Page Title
    </h1>
    <p
      className="fd-action-bar__description"
    >
      Action Bar Description
       
    </p>
  </div>
  <div
    className="fd-action-bar__actions blue"
  >
    <button>
      Button
    </button>
  </div>
</div>
`;

exports[`<ActionBar /> create basic mobile Action Bar 1`] = `
<div
  style={
    Object {
      "width": "319px",
    }
  }
>
  <div
    className="fd-action-bar blue"
  >
    <div
      className="fd-action-bar__back"
    >
      <button
        className="fd-button--light fd-button--compact sap-icon--nav-back"
      />
    </div>
    <div
      className="fd-action-bar__header"
    >
      <h1
        className="fd-action-bar__title"
      >
        Page Title
      </h1>
      <p
        className="fd-action-bar__description"
      >
        Action Bar Description
         
      </p>
    </div>
    <div
      className="fd-action-bar__actions"
    >
      <button>
        Button
      </button>
    </div>
  </div>
</div>
`;

exports[`<ActionBar /> create basic mobile Action Bar 2`] = `
<div
  style={
    Object {
      "width": "500px",
    }
  }
>
  <div
    className="fd-action-bar"
  >
    <div
      className="fd-action-bar__back"
    >
      <button
        className="fd-button--light fd-button--compact sap-icon--nav-back"
      />
    </div>
    <div
      className="fd-action-bar__header"
    >
      <h1
        className="fd-action-bar__title"
      >
        Page Title
      </h1>
      <p
        className="fd-action-bar__description"
      >
        Action Bar Description
         
      </p>
    </div>
    <div
      className="fd-action-bar__actions"
    >
      <button>
        Button
      </button>
    </div>
  </div>
</div>
`;
